File: BinaryTreeLevelOrderTraversal/fourth.cpp
void traverse(TreeNode* node, vector<vector<int>> &result, int level){
        if(!node) return;
        if (result.size() < level+1)
            result.push_back(vector<int>());
        result[level++].push_back(node->val);
        traverse(node->left, result, level);
        traverse(node->right, result, level);
    }
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> result;
        traverse(root, result, 0);
        return result;
    }

File: BinaryTreeLevelOrderTraversal/fourth.py
def levelOrder(self, root):
    if root is None:
        return []
    q = [root]
    output = []
    while len(q):
        level = []
        for x in q:
            level.append(x.val)
        output.append(level)
        q2=[]
        for x in q:
            if x.left is not None:
                q2.append(x.left)
            if x.right is not None:
                q2.append(x.right)
        q = q2
    return output
